### PR TITLE
Add MSA write gazebo urdf feature

### PR DESCRIPTION
**Pull request by rickstaa**
### Description

This pull request adds several improvements to the simulation tab of the MSA. The most significant change is that the Gazebo URDF shown in the simulation tab was not saved in the old version. As a result, when the provided URDF was incompatible with the Gazebo, the `gazebo.launch` file was not working. The new version now ensures that a gazebo URDF is saved as a MoveIT configuration file when the provided URDF is invalid. It further provides users with the option to overwrite the provided URDF and fixes a small bug:

- [fix(msa): fix redunant transmission tag create bug](https://github.com/ros-planning/moveit/pull/2960/commits/74f7bd3f4c7ad60af5f7ac8f62ee85743ff007ae)

I first tried 22284a59f6bc8c3d7a138681c261f04df009d286. However, in the end, I added 74f7bd3f4c7ad60af5f7ac8f62ee85743ff007ae since it does a better job at checking whether a urdf is Gazebo compatible. 

#### New configuration no sim

new_now_sim

The `gazebo_panda.urdf` is **NOT**  added.

#### New configuration + sim + compatible xacro

Here I use https://github.com/frankaemika/franka_ros/pull/199 for the xacro description. This model is Gazebo compatible. I however had to add an inertial tag to the [panda_link8_sc link](https://github.com/frankaemika/franka_ros/blob/63ecb779ae49d023b5ef93434a457bb599c553ed/franka_description/robots/arm.xacro#L271-L289) since my script has no knowledge of why this link has no inertia.

```diff
<link name="${prefix}link8_sc">
      <collision>
        <origin xyz="0.0424 0.0424 -0.0250" rpy="${pi} ${pi/2} ${pi/2}"/>
        <geometry>
          <cylinder radius="${0.03+safety_distance}"  length="0.01" />
        </geometry>
      </collision>
      <collision>
        <origin xyz="0.0424 0.0424 -0.02" rpy="0 0 0"/>
        <geometry>
          <sphere radius="${0.03+safety_distance}"  />
        </geometry>
      </collision>
      <collision>
        <origin xyz="0.0424 0.0424 -0.03" rpy="0 0 0"/>
        <geometry>
          <sphere radius="${0.03+safety_distance}"  />
        </geometry>
      </collision>
+      <inertial>
+        <mass value="0"/>
+       <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0" />
+     </inertial>
    </link>
```

new_gazebo_compatibe_xacro

The `gazebo_panda.urdf` is **NOT**  added.

#### New configuration + sim + xacro description

new_gazebo_incompatible_xacro gif

The `gazebo_panda.urdf` is added.

#### New configuration + sim + xacro description + overwrite

new_gazebo_incompatible_xacro_overwrite

The `gazebo_panda.urdf` is added.

#### New configuration + sim + compatible urdf

new_gazebo_compatible_urdf

The `gazebo_panda.urdf` is **NOT**  added.

#### New configuration + sim + urdf description

new_gazebo_incompatible_urdf

The `gazebo_panda.urdf` is added.

#### New configuration + sim + urdf description + overwrite

new_gazebo_incompatible_urdf_overwrite

The `gazebo_panda.urdf` is **NOT**  added.

#### Existing configuration no sim

existing_no_sim

The `gazebo_panda.urdf` is **NOT**  added.

#### Existing configuration + sim + xacro description

existing_no_sim_xacro

The `gazebo_panda.urdf` is added.

#### Existing configuration + sim + xacro description + overwrite

existing_no_sim_xacro_overwrite

The `gazebo_panda.urdf` is added.

#### Existing configuration + sim + compatible urdf description

existing_sim_compatible_urdf

The `gazebo_panda.urdf` is **NOT** added.

#### Existing configuration + sim + urdf description

existing_sim_incompatible_urdf

The `gazebo_panda.urdf` is added.

#### Existing configuration + sim + urdf description + overwrite

existing_sim_incompatible_urdf_overwrite

The `gazebo_panda.urdf` is **NOT** added.

### Checklist
- [x] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [x] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
    - [x] Simulation section of the moveit_tutorials have to be updated will be handled in https://github.com/ros-planning/moveit_tutorials/pull/678
- [x] Include a screenshot if changing a GUI

### Other downstream TODOS
- [ ] If this pull request is merged open a pull request at [franka_ros](https://github.com/frankaemika/franka_ros) to add a dummy `inertial` tag to the `panda_link8_sc` link. This way the Franka `urdf` is seen as Gazebo compatible.

### CODE REVIEW

File: moveit_setup_assistant/src/tools/moveit_config_data.cpp [hunk 1]
```diff
@@ -443,11 +452,14 @@ std::string MoveItConfigData::getGazeboCompatibleURDF()
       }
       else if (static_cast<std::string>(doc_element->Value()).find("joint") != std::string::npos)
       {
-        // Before adding a transmission element, make sure there the joint is not fixed
-        if (static_cast<std::string>(doc_element->Attribute("type")) != "fixed")
+        // Before adding a transmission element, make sure there is none and the joint is not fixed
+        std::string joint_name = static_cast<std::string>(doc_element->Attribute("name"));
+        if (static_cast<std::string>(doc_element->Attribute("type")) != "fixed" &&
+            !std::any_of(trans_elements.begin(), trans_elements.end(), [joint_name](const std::string& trans) {
```
Comment (rickstaa) [inline moveit_setup_assistant/src/tools/moveit_config_data.cpp:482]: Has to be improved doesn't work when the name is different.

Comment (rickstaa) [reply moveit_setup_assistant/src/tools/moveit_config_data.cpp]: I improved upon this by using https://github.com/ros-planning/moveit/pull/2960/commits/74f7bd3f4c7ad60af5f7ac8f62ee85743ff007ae.

**Comment by rhaschke** [comment]
@rickstaa, sorry that your PR was ignored for so long time! To resolve the merge conflicts and facilitate the review process, I rebased your changes to master and augmented my suggestions directly as additional commits. Please have a look at #3081. Closing here.

**Comment by rickstaa** [comment]
@rhaschke No problem. Thanks a lot for working on this!

**Pull request closed without merging**